### PR TITLE
JS page types, part 1

### PR DESCRIPTION
**Pull request by wbamberg**
Part of https://github.com/mdn/content/issues/16156.

I'm doing these in batches to reduce review pain. This is I expect the first of four.

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/reflect/comparing_reflect_and_object_methods/index.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Comparing Reflect and Object methods
 slug: >-
+page-type: guide
   Web/JavaScript/Reference/Global_Objects/Reflect/Comparing_Reflect_and_Object_methods
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/global_objects/reflect/comparing_reflect_and_object_methods/index.md:5]:
```suggestion
slug: >-
  Web/JavaScript/Reference/Global_Objects/Reflect/Comparing_Reflect_and_Object_methods
page-type: guide
```

File: files/en-us/web/javascript/reference/errors/delete_in_strict_mode/index.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 title: >-
   SyntaxError: applying the 'delete' operator to an unqualified name is
+page-type: javascript-error
   deprecated
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/errors/delete_in_strict_mode/index.md:5]:
```suggestion
  deprecated
page-type: javascript-error
```

File: files/en-us/web/javascript/reference/errors/deprecated_source_map_pragma/index.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ---
 title: >-
   SyntaxError: Using //@ to indicate sourceURL pragmas is deprecated. Use //#
+page-type: javascript-error
   instead
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/errors/deprecated_source_map_pragma/index.md:5]:
```suggestion
  instead
page-type: javascript-error
```

File: files/en-us/web/javascript/reference/errors/invalid_for-of_initializer/index.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ---
 title: >-
   SyntaxError: a declaration in the head of a for-of loop can't have an
+page-type: javascript-error
   initializer
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/errors/invalid_for-of_initializer/index.md:5]:
```suggestion
  initializer
page-type: javascript-error
```

File: files/en-us/web/javascript/reference/global_objects/finalizationregistry/finalizationregistry/index.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ---
 title: FinalizationRegistry() constructor
 slug: >-
+page-type: javascript-constructor
   Web/JavaScript/Reference/Global_Objects/FinalizationRegistry/FinalizationRegistry
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/global_objects/finalizationregistry/finalizationregistry/index.md:5]:
```suggestion
  Web/JavaScript/Reference/Global_Objects/FinalizationRegistry/FinalizationRegistry
page-type: javascript-constructor
```

File: files/en-us/web/javascript/reference/global_objects/intl/relativetimeformat/relativetimeformat/index.md [hunk 6]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Intl.RelativeTimeFormat() constructor
 slug: >-
+page-type: javascript-constructor
   Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat/RelativeTimeFormat
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/global_objects/intl/relativetimeformat/relativetimeformat/index.md:5]:
```suggestion
  Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat/RelativeTimeFormat
page-type: javascript-constructor
```

File: files/en-us/web/javascript/reference/errors/deprecated_source_map_pragma/index.md [hunk 7]
```diff
@@ -2,6 +2,7 @@
 title: >-
   SyntaxError: Using //@ to indicate sourceURL pragmas is deprecated. Use //#
   instead
+page-type: javascript-error
 slug: Web/JavaScript/Reference/Errors/Deprecated_source_map_pragma
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/errors/deprecated_source_map_pragma/index.md:6]:
```suggestion
slug: Web/JavaScript/Reference/Errors/Deprecated_source_map_pragma
page-type: javascript-error
```

File: files/en-us/web/javascript/reference/errors/invalid_for-of_initializer/index.md [hunk 8]
```diff
@@ -2,6 +2,7 @@
 title: >-
   SyntaxError: a declaration in the head of a for-of loop can't have an
   initializer
+page-type: javascript-error
 slug: Web/JavaScript/Reference/Errors/Invalid_for-of_initializer
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/errors/invalid_for-of_initializer/index.md:6]:
```suggestion
slug: Web/JavaScript/Reference/Errors/Invalid_for-of_initializer
page-type: javascript-error
```

**Review by Josh-Cena — APPROVED** [review]
This is EZ :D

**Pull request merged**